tests/cpp/test_dynamic_size.cpp

```diff
@@ -124,4 +124,137 @@ TEST(Converters, ATenResizeGetItemDynShapeMulCorrectly) {
   auto trt = trt_results[0].reshape(jit_results[0].sizes());
 
   ASSERT_TRUE(torch_tensorrt::tests::util::almostEqual(jit_results[0], trt, 2e-6));
-}
\ No newline at end of file
+}
+
+TEST(Converters, ATenUnflattenDynShapeShapeCorrectly) {
+  const auto graph = R"IR(
+    graph(%x.1 : Tensor):
+            %2 : int = prim::Constant[value=1]()
+            %3 : int = prim::Constant[value=512]()
+            %4 : int = prim::Constant[value=1]()
+            %5 : int = prim::Constant[value=1]()
+            %6 : int[] = prim::ListConstruct(%3, %4, %5)
+            %7 : Tensor = aten::unflatten(%x.1, %2, %6)
+            return (%7))IR";
+
+  auto g = std::make_shared<torch::jit::Graph>();
+
+  torch::jit::parseIR(graph, g.get());
+
+  auto in = at::randint(0, 10, {1, 512}, {at::kCUDA});
+
+  auto jit_in = at::clone(in);
+  auto params = torch_tensorrt::core::ir::get_static_params(g->inputs(), {});
+  auto jit_results = torch_tensorrt::tests::util::RunGraph(g, params, {jit_in});
+
+  auto trt_in = at::clone(in);
+  params = torch_tensorrt::core::ir::get_static_params(g->inputs(), {});
+  auto trt_results = torch_tensorrt::tests::util::RunGraphEngineDynamic(g, params, {in}, true);
+
+  ASSERT_TRUE(torch_tensorrt::tests::util::almostEqual(jit_results[0], trt_results[0], 2e-6));
+}
+
+TEST(Converters, ATenUnflattenDynShapeNegativeDimsShapeCorrectly) {
+  const auto graph = R"IR(
+    graph(%x.1 : Tensor):
+            %2 : int = prim::Constant[value=-2]()
+            %3 : int = prim::Constant[value=512]()
+            %4 : int = prim::Constant[value=1]()
+            %5 : int = prim::Constant[value=1]()
+            %6 : int[] = prim::ListConstruct(%3, %4, %5)
+            %7 : Tensor = aten::unflatten(%x.1, %2, %6)
+            return (%7))IR";
+
+  auto g = std::make_shared<torch::jit::Graph>();
+
+  torch::jit::parseIR(graph, g.get());
+
+  auto in = at::randint(0, 10, {1, 512, 2}, {at::kCUDA});
+
+  auto jit_in = at::clone(in);
+  auto params = torch_tensorrt::core::ir::get_static_params(g->inputs(), {});
+  auto jit_results = torch_tensorrt::tests::util::RunGraph(g, params, {jit_in});
+
+  auto trt_in = at::clone(in);
+  params = torch_tensorrt::core::ir::get_static_params(g->inputs(), {});
+  auto trt_results = torch_tensorrt::tests::util::RunGraphEngineDynamic(g, params, {in}, true);
+
+  ASSERT_TRUE(torch_tensorrt::tests::util::almostEqual(jit_results[0], trt_results[0], 2e-6));
+}
```

Comment:
Can you move these to test_shuffle.cpp since these are not ITensor based (aten::size + dyn_shapes) ? The aten::size + aten::unflatten.cpp tests can be left in this file.

Reply:
Removed redundant test cases. Mentioned test case variants are already present in test_shuffle.